include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -271,8 +300,50 @@ namespace RINGMesh
 
         GEO::AttributesManager attributes_manager_;
     };
+    ALIAS_2D_AND_3D( CartesianGridBase );
+
+    template < index_t DIMENSION >
+    class CartesianGrid final : public CartesianGridBase< DIMENSION >
+    {
+        friend class CartesianGridBuilder< DIMENSION >;
```

Comment:
Default constructor is private in a class. So, do you want to forbid your class in 2D?